File: 백준/Silver/18258. 큐 2/큐 2.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        Queue<Integer> queue = new LinkedList<>();
        
        int n = Integer.parseInt(br.readLine());
        
        for (int i = 0; i < n; i++) {
            String line = br.readLine();
            String[] command = line.split(" ");
            
            if (command[0].equals("push")) {
                queue.add(Integer.parseInt(command[1]));
            } else if (command[0].equals("pop")) {
                if (queue.isEmpty()) {
                    sb.append(-1).append('\n');
                } else {
                    sb.append(queue.poll()).append('\n');
                }
            } else if (command[0].equals("size")) {
                sb.append(queue.size()).append('\n');
            } else if (command[0].equals("empty")) {
                if (queue.isEmpty()) {
                    sb.append(1).append('\n');
                } else {
                    sb.append(0).append('\n');
                }
            } else if (command[0].equals("front")) {
                if (queue.isEmpty()) {
                    sb.append(-1).append('\n');
                } else {
                    sb.append(queue.peek()).append('\n');
                }
            } else if (command[0].equals("back")) {
                if (queue.isEmpty()) {
                    sb.append(-1).append('\n');
                } else {
                    sb.append(((LinkedList<Integer>) queue).peekLast()).append('\n');
                }
            }
        }
        
        System.out.print(sb.toString());
    }
}

File: 백준/Silver/18258. 큐 2/큐 2.py
import sys
from collections import deque

input = sys.stdin.readline

queue = deque()
results = []

while True:
    line = input().strip()
    if not line:
        break
    command = line.split()
    
    if command[0] == "push":
        queue.append(int(command[1]))
    elif command[0] == "pop":
        if queue:
            results.append(queue.popleft())
        else:
            results.append(-1)
    elif command[0] == "size":
        results.append(len(queue))
    elif command[0] == "empty":
        results.append(1 if not queue else 0)
    elif command[0] == "front":
        results.append(queue[0] if queue else -1)
    elif command[0] == "back":
        results.append(queue[-1] if queue else -1)

# 출력 결과를 줄 단위로 작성
sys.stdout.write("\n".join(map(str, results)) + "\n")
